src/modules/Dropdown/Dropdown.js

```diff
@@ -227,6 +246,25 @@ export default class Dropdown extends Component {
       debug('value changed, setting', nextProps.value)
       this.setValue(nextProps.value)
     }
+
+    if (!_.isEqual(nextProps.options, this.props.options)) {
+      if (_.first(this.state.options) !== _.first(this.props.options)) {
+        // `Add: ...` option at the top of the list
+        this.setState({
+          options: [_.first(this.state.options), ...nextProps.options],
+        })
+      } else if (_.last(this.state.options) !== _.last(this.props.options)) {
+        // `Add: ...` option at the bottom of the list
+        this.setState({
+          options: [...nextProps.options, _.last(this.state.options)],
+        })
+      } else {
+        // No `Add: ...` option -> just copy the list
+        this.setState({
+          options: [...nextProps.options],
+        })
```

Comment:
Note, this also resovles the bug I noted about backspacing a query.  That is because the logic here includes `searchQuery` as a condition for injecting the `add` item.